packages/task/src/browser/task-service.ts

```diff
@@ -237,14 +239,41 @@ export class TaskService implements TaskConfigurationClient {
      * It looks for configured and provided tasks.
      */
     async run(source: string, taskLabel: string): Promise<void> {
+        const task = await this.getTaskConfiguration(source, taskLabel);
+        if (!task) {
+            this.logger.error(`Can't get task launch configuration for label: ${taskLabel}`);
+            return;
+        }
+
+        this.runTask(task);
+    }
+
+    protected async getTaskConfiguration(source: string, taskLabel: string): Promise<TaskConfiguration | undefined> {
         let task = await this.getProvidedTask(source, taskLabel);
         if (!task) {
             task = this.taskConfigurations.getTask(source, taskLabel);
-            if (!task) {
-                this.logger.error(`Can't get task launch configuration for label: ${taskLabel}`);
-                return;
-            }
         }
+        return task;
+    }
+
+    /**
+     * The same as run method, but the last parameter may contain additional task type specific properties
+     * which are used only for this request over persistent configuration of the task.
+     */
+    // tslint:disable-next-line:no-any
+    async customRun(source: string, taskLabel: string, taskPropertiesOverrides?: { [key: string]: any }): Promise<void> {
+        if (!taskPropertiesOverrides) {
+            this.run(source, taskLabel);
+            return;
+        }
+
+        let task = await this.getTaskConfiguration(source, taskLabel);
+        if (!task) {
+            this.logger.error(`Can't get task launch configuration for label: ${taskLabel}`);
+            return;
+        }
+
+        task = <TaskConfiguration>deepmerge([task, taskPropertiesOverrides]);
```

Comment:
I guess introducing a new library requires a CQ. So, maybe just use
```js
Object.assign({}, task, taskPropertiesOverrides)
```
Will it solve the problem?

Reply:
> I guess introducing a new library requires a CQ.

No, this lib is already used in Theia

> Object.assign({}, task, taskPropertiesOverrides)
Will it solve the problem?

No again. We do not know how task options are organized (think of custom task provider). `Object.assign` will replace only top level properties. If original configuration has object as a property and amend has changes only for one field of that property - then `Object,assign` will fail.

Reply:
I see. Thanks for the explanation.